### ISSUE TITLE
Ignore not working

### ISSUE THREAD

**Issue by mattdell** [nature/pa11y#67]
Hello,

Hopefully I'm just doing something wrong but I can't get the ignore feature to work. Here is my code:

	pa11y.sniff({
		url: url,
		standard: accessibilityStandard,
		useragent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.104 Safari/537.36',
		config: {
			ignore: [
				'WCAG2AA.Principle1.Guideline1_4.1_4_3.G18'
			]
		},
	}, function(err, results) {
		// Do something
	});

Whenever I run this command I get nothing but WCAG2AA.Principle1.Guideline1_4.1_4_3.G18 errors (because I've eliminated everything else).

Is there something wrong in my code or is there a defect?

**Comment by whymarrh** [comment]
Hey Matt, can you share the url and standard you are running the test with (I'm assuming from the error that it's WCAG2AA)? Note that if you are trying to ignore `WCAG2AA.`-prefixed errors, that you need to be testing against `WCAG2AA`.

**Comment by whymarrh** [comment]
Just as an example, the following works for me, and running it with and without the ignore produce the expected results:

```js
var pa11y = require('pa11y');
pa11y.sniff({
    url: 'https://www.reddit.com/r/thebutton/',
    standard: 'WCAG2A',
    useragent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.104 Safari/537.36',
    config: {
        ignore: [
            'WCAG2A.Principle2.Guideline2_4.2_4_4.H77,H78,H79,H80,H81'
        ]
    },
}, function(err, results) {
    if (err) {
        console.error(err);
        process.exit(1);
    }
    console.log(results);
});
```

**Comment by mattdell** [comment]
Hey, thanks for the quick response!

I'm running it against localhost so I can't share the URL. But, it seems changing the ignore rule from `WCAG2AA.Principle1.Guideline1_4.1_4_3.G18` to `WCAG2AA.Principle1.Guideline1_4.1_4_3.G18.Fail` seems to have worked.

I would have thought it would work without the .Fail.

**Comment by whymarrh** [comment]
Yeah, it requires the rule verbatim as it appears in the results. Maybe it should work w/o the `.Fail` (and possibly without the standard prefix)? That would be a good enhancement.

**Comment by mattdell** [comment]
I think a good enhancement would be to have the ignore rules in sync with this list: https://github.com/nature/pa11y/wiki/HTML-CodeSniffer-Rules

That's where I got the rule that I wanted to ignore and then copy/pasted it from there. I hadn't thought of copy/pasting straight from terminal.

**Comment by whymarrh** [comment]
Well it did say that they "can be used in a config file to ignore certain rules", so...yeah. I've updated that page to try and clarify the workflow -- some of those rules are dependent on e.g. `[NodeName]` so I'm not sure if it could be in sync.

**Comment by mattdell** [comment]
Cool, thanks for looking into that. Hopefully it helps clear it up for anyone else!
